provider/cldr/src/transform/datetime/mod.rs

```diff
@@ -2,8 +2,320 @@
+
+                        let cal_file = format!("ca-{}.json", cldr_cal);
+                        let path = locale_dir.join(&cal_file);
+
+                        let mut resource: cldr_serde::ca::Resource =
+                            serde_json::from_reader(open_reader(&path)?)
+                                .map_err(|e| Error::Json(e, Some(path)))?;
+
+                        self.data.insert(req.options.clone(), Box::new(resource
+                            .main
+                            .0
+                            .remove(langid)
+                            .expect("CLDR file contains the expected language")
+                            .dates
+                            .calendars
+                            .remove(*cldr_cal)
+                            .ok_or_else(|| {
+                                Error::Custom(
+                                    format!("{} does not have {} field", cal_file, cldr_cal),
+                                    None,
+                                )
+                        })?))
+                    };
+
+                    let metadata = DataResponseMetadata::default();
+                    // TODO(#1109): Set metadata.data_langid correctly.
+                    Ok(DataResponse {
+                        metadata,
+                        #[allow(clippy::redundant_closure_call)]
+                        payload: Some(DataPayload::from_owned(($expr)(dates, variant))),
+                    })
+                }
+            }
+
+            impl IterableResourceProvider<$marker> for CommonDateProvider {
+                fn supported_options(
+                    &self,
+                ) -> Result<Box<dyn Iterator<Item = ResourceOptions> + '_>, DataError> {
+                    // Raise an error if Gregorian paths are not available
+                    self.paths
+                        .iter()
+                        .find(|(_, cal, _)| cal == &"gregory")
+                        .ok_or(Error::MissingSource(crate::error::MissingSourceError {
+                            src: "cldr-dates/gregory",
+                        }))?;
+
+                    let mut r = Vec::new();
+                    for (_, cal, path) in &self.paths {
+                        let cal = Some((*cal).into());
+                        r.extend(
+                            get_langid_subdirectories(&path.join("main"))?
+                                .map(Into::<ResourceOptions>::into)
+                                .map(move |mut r| {
+                                    r.variant = cal.clone();
+                                    r
+                                }),
+                        );
+                    }
+                    Ok(Box::new(r.into_iter()))
+                }
+            }
+        )+
+
+        icu_provider::impl_dyn_provider!(CommonDateProvider, [$($marker),+,], SERDE_SE);
```

Comment:
note: i think with the verification PR landed this needs to be a single invocation outside with the three markers

Reply:
oh I see

Reply:
It still needs `impl ConvertData` after you rebase, though

Reply:
done